avm/res/sql/server/database/main.bicep

```
@@ -2,146 +2,148 @@ metadata name = 'SQL Server Database'
+  | 'Recovery'
+  | 'Restore'
+  | 'RestoreExternalBackup'
+  | 'RestoreExternalBackupSecondary'
+  | 'RestoreLongTermRetentionBackup'
+  | 'Secondary' = 'Default'
+
+@description('Optional. The resource ID of the elastic pool containing this database.')
+param elasticPoolId string?
 
-@description('Optional. The name of the SKU.')
-param skuName string = 'GP_Gen5_2'
+@description('Optional. The azure key vault URI of the database if it\'s configured with per Database Customer Managed Keys.')
+param encryptionProtector string?
 
-@description('Optional. Capacity of the particular SKU.')
-param skuCapacity int?
+@description('Optional. The flag to enable or disable auto rotation of database encryption protector AKV key.')
+param encryptionProtectorAutoRotation bool?
 
-@description('Optional. Type of enclave requested on the database i.e. Default or VBS enclaves.')
-@allowed([
-  ''
-  'Default'
-  'VBS'
-])
-param preferredEnclaveType string = ''
+@description('Optional. The Client id used for cross tenant per database CMK scenario.')
+@minLength(36)
+@maxLength(36)
+param federatedClientId string?
+
+@description('Optional. Specifies the behavior when monthly free limits are exhausted for the free database.')
+param freeLimitExhaustionBehavior 'AutoPause' | 'BillOverUsage'?
+
+@description('Optional. The number of readonly secondary replicas associated with the database.')
+param highAvailabilityReplicaCount int = 0
 
-@description('Optional. If the service has different generations of hardware, for the same SKU, then that can be captured here.')
-param skuFamily string = ''
+@description('Optional. Whether or not this database is a ledger database, which means all tables in the database are ledger tables. Note: the value of this property cannot be changed after the database has been created.')
+param isLedgerOn bool = false
+
+@description('Optional. The license type to apply for this database.')
+param licenseType 'BasePrice' | 'LicenseIncluded'?
+
+@description('Optional. The resource identifier of the long term retention backup associated with create operation of this database.')
+param longTermRetentionBackupResourceId string?
+
+@description('Optional. Maintenance configuration ID assigned to the database. This configuration defines the period when the maintenance updates will occur.')
+param maintenanceConfigurationId string?
 
-@description('Optional. Size of the particular SKU.')
-param skuSize string = ''
+@description('Optional. Whether or not customer controlled manual cutover needs to be done during Update Database operation to Hyperscale tier.')
+param manualCutover bool?
 
 @description('Optional. The max size of the database expressed in bytes.')
 param maxSizeBytes int = 34359738368
 
-@description('Optional. The name of the sample schema to apply when creating this database.')
-param sampleName string = ''
+@description('Optional. Minimal capacity that database will always have allocated.')
+param minCapacity string = '0'
```

```suggestion
param minCapacity int = 0
```
as per [spec](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr14---category-inputs---data-types). We both know the Resource Provider won't like that, so the casting must happen where the value is used (e.g., `'${minCapacity}'`)

---

`int` would not work. If you define as `int`, you cannot pass in fractional values, like 0.5 or 0.25, which is needed for this param.

Even the [API definition](https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/servers/databases?pivots=deployment-language-bicep) is saying: `minCapacity: json('decimal-as-string')`

---

Hey @peterbud, thanks for elaburating. Then that's my bad - I was under the impression it is a full number, not a decimal. Resolving